privacyidea/static/components/directives/views/directive.tokendata.html

```diff
@@ -10,24 +10,27 @@
     </form>
 </td>
 
-<td><div ng-hide="loggedInUser.role == 'user' || tokenLocked == 'true' ">
-    <button ng-show="editFlag"
-            class="btn btn-primary"
-            ng-click="callback({key: tokenKey, value: tokenData}); editFlag =
-            false;"
-            ng-disabled="tokenDataForm.$invalid"
-            >{{ buttonText }}
-    </button>
-    <button class="btn btn-danger"
-            ng-show="editFlag"
-            ng-click="callbackCancel(); editFlag = false;"
-            translate>Cancel
-    </button>
-    <button ng-hide="editFlag || (setRight == 'false' && hideButtons == 'true')"
-            class="btn btn-primary"
-            ng-click="editFlag = true;"
-            ng-disabled="setRight == 'false'"
-            translate>Edit
-    </button>
+<td>
+    <div ng-hide="loggedInUser.role == 'user' && editableAsUser != 'true' ">
```

Comment:
To make the logic work more reliable I split this in two div's.
If a directive is ediable for a user, we will *not* hide the edit button!